utils/doclint/generateDotnetApi.js

```diff
@@ -0,0 +1,695 @@
+    return `${parent.name}${translateMemberName(member.kind, member.name, null)}Result`;
+  });
+
+  /** @type {Map<string, string[]>} */
+  const paramDocs = new Map();
+  const addParamsDoc = (paramName, docs) => {
+    if (paramName.startsWith('@'))
+      paramName = paramName.substring(1);
+    if (paramDocs.get(paramName))
+      throw `Parameter ${paramName} already exists in the docs.`;
+    paramDocs.set(paramName, docs);
+  };
+
+  /** @type {string} */
+  let type = null;
+  // need to check the original one
+  if (member.type.name === 'Object' || member.type.name === 'Array') {
+    let innerType = member.type;
+    let isArray = false;
+    if (innerType.name === 'Array') {
+      // we want to influence the name, but also change the object type
+      innerType = member.type.templates[0];
+      isArray = true;
+    }
+
+    if (innerType.expression === '[Object]<[string], [string]>') {
+      // do nothing, because this is handled down the road
+    } else if (!isArray && !innerType.properties) {
+      type = `dynamic`;
+    } else {
+      type = classNameMap.get(innerType.name);
+      if (!type) {
+        type = typeResolve(innerType);
+        // additionalTypes.set(type, innerType);
+      }
+
+      if (isArray)
+        type = `IReadOnlyCollection<${type}>`;
+    }
+  }
+
+  type = type || typeResolve(member.type); //translateType(member.type, parent);
+  // TODO: this is something that will probably go into the docs
+  if (member.args.size == 0
+    && type !== 'void'
+    && !name.startsWith('Is')
+    && !name.startsWith('Get')) {
+    if (!member.async) {
+      if (member.spec)
+        output(XmlDoc.renderXmlDoc(member.spec, maxDocumentationColumnWidth).map(x => `\t${x}`));
+      output(`${type} ${name} { get; }`);
+      return;
+    }
+    name = `Get${name}`;
+  }
+
+  // HACK: special case for generics handling!
+  if (type === 'T') {
+    name = `${name}<T>`;
+  }
+
+  // adjust the return type for async methods
+  if (member.async)
+    if (type === 'void')
+      type = `Task`;
+    else
+      type = `Task<${type}>`;
+
+  // render args
+  let args = [];
+  /**
+   * 
+   * @param {string} innerArgType 
+   * @param {string} innerArgName 
+   * @param {Documentation.Member} argument 
+   */
+  const pushArg = (innerArgType, innerArgName, argument) => {
+    let isEnum = enumTypes.has(innerArgType);
+    let isNullable = ['int', 'bool', 'decimal'].includes(innerArgType);
+    const requiredPrefix = argument.required ? "" : isNullable ? "?" : "";
+    const requiredSuffix = argument.required ? "" : isEnum ? " = default" : " = null";
+    args.push(`${innerArgType}${requiredPrefix} ${innerArgName}${requiredSuffix}`);
+  };
+
+  let parseArg = (/** @type {Documentation.Member} */ arg) => {
+    if (arg.name === "options") {
+      arg.type.properties.forEach(prop => {
+        parseArg(prop);
+      });
+      return;
+    }
+
+    if (arg.type.expression === '[string]|[path]') {
```

Comment:
Can we define this as overloaded params in the documentation? I don't quite understand what should happen with this logic in this case of setInputFiles files parameter ( <[path]|[Array]<[path]>|[Object]|[Array]<[Object]>>).